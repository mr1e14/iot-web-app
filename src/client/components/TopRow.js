import React from "react";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import WeatherWidget from "./WeatherWidget";

const TopRow = props => (
  <Grid
    container
    direction="row"
    alignItems="flex-start"
    justify="space-between"
  >
    <Grid className={props.classes.child}>
      <WeatherWidget {...props} />
    </Grid>
    <Grid className={props.classes.child}>
      <Button
        variant="outlined"
        classes={{ label: props.classes.buttonLabel }}
        size="large"
        color="secondary"
      >
        <Typography color="textPrimary">Sign in</Typography>
      </Button>
    </Grid>
  </Grid>
);

export default TopRow;
