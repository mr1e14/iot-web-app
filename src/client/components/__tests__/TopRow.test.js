import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

jest.mock("../mediaQuery");

import TopRow from "../TopRow";

const classes = {
  child: "child",
  childGridCenter: "childGridCenter",
  childGridLeft: "childGridLeft",
  childGridRight: "childGridRight",
  buttonLabel: "buttonLabel"
};

const weatherData = {
  icon: "partly-cloudy",
  temperature: "15",
  humidity: "0.75"
};

const exampleProps = {
  classes,
  weatherData
};

describe("TopRow", () => {
  let tree;
  let component;
  beforeEach(() => {
    component = renderer.create(<TopRow {...exampleProps} />);
    tree = component.toJSON();
  });
  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });
  describe("when renders", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(<TopRow {...exampleProps} />);
    });
    it("has three Grid components", () => {
      expect(wrapper.find("Grid").length).toBe(3);
    });
    it("has a container Grid", () => {
      const { direction, alignItems, justify } = wrapper
        .find("Grid")
        .at(0)
        .props();
      expect(direction).toEqual("row");
      expect(alignItems).toEqual("flex-start");
      expect(justify).toEqual("space-between");
    });
    it("has Grid with WeatherWidget", () => {
      expect(
        wrapper
          .find("Grid")
          .at(1)
          .find("WeatherWidget").length
      ).toBe(1);
    });
    it("has Grid with 'sign in' Button", () => {
      expect(
        wrapper
          .find("Grid")
          .at(2)
          .find("Button")
          .find("Typography")
          .text()
      ).toBe("Sign in");
    });
  });
});
