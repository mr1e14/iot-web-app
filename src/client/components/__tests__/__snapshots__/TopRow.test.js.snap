// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRow renders correctly 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-align-items-xs-flex-start-10 MuiGrid-justify-xs-space-between-20"
>
  <div
    className="childGridLeft"
  >
    <span
      className="MuiTypography-root-98 MuiTypography-caption-108"
    >
      <img
        align="middle"
        alt="weather-icon"
        height="48px"
        style={
          Object {
            "marginRight": "0.5rem",
          }
        }
      />
      <span>
        15Â°C
      </span>
    </span>
  </div>
  <div
    className="childGridRight"
  >
    <button
      className="MuiButtonBase-root-160 MuiButton-root-134 MuiButton-outlined-142 MuiButton-outlinedPrimary-143 MuiButton-sizeLarge-158"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-135 buttonLabel"
      >
        <p
          className="MuiTypography-root-98 MuiTypography-body1-107 MuiTypography-colorTextPrimary-130"
        >
          Sign in
        </p>
      </span>
      <span
        className="MuiTouchRipple-root-163"
      />
    </button>
  </div>
</div>
`;
